### DISCUSSION TITLE
No Mileage Shown Corresponding to Software Updates

### DISCUSSION THREAD

**Discussion by SemoTech** [teslamate-org/teslamate#2398] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

While reviewing the "updates" data showing all software updates the car has received, the mileage on the vehicle at the time of the update installation is not showing.  Can that be enabled/added?

### Expected Behavior

Ability to see Mileage at the time of each vehicle software update.

### Steps To Reproduce

In Dashboard, Select "Updates"

### Relevant log output

```shell
n/a
```

### Screenshots

n/a

### Type of installation

Docker

### Version

1.26.0

**Comment by cwanja** [comment]
Not shooting this down, but curious - what is the benefit here? 

Back dating the data would be challenging. Doable, but challenging. Although, I am not a scripting genius by any stretch of the imagination.

**Comment by SemoTech** [comment]
> Not shooting this down, but curious - what is the benefit here?
> 
> Back dating the data would be challenging. Doable, but challenging. Although, I am not a scripting genius by any stretch of the imagination.

Hello @cwanja, the entire point of this amazing platform is to gather as much useful data about the vehicle as possible.  Storing the mileage of the vehicle, at the time of each software updates, is another data point that is useful when looking at the history of the software updates. For now I have to maintain a separate manually updated list so that I know at what mileage I received what features and improvements and would love for that to be part of Teslamate.

**Comment by cwanja** [comment]
> > Not shooting this down, but curious - what is the benefit here?
> > Back dating the data would be challenging. Doable, but challenging. Although, I am not a scripting genius by any stretch of the imagination.
> 
> Hello @cwanja, the entire point of this amazing platform is to gather as much useful data about the vehicle as possible. Storing the mileage of the vehicle, at the time of each software updates, is another data point that is useful when looking at the history of the software updates. For now I have to maintain a separate manually updated list so that I know at what mileage I received what features and improvements and would love for that to be part of Teslamate.

Sounds good. Definitely seems doable, considering there is a mileage dashboard. Could take the time from the update row and extract the stored mileage from that point. @DrMichael, pinging you for your SQL skillset :)

**Comment by sunflower-green** [comment]
In my opinion I can't really see the benefit here. Why do I want to know the mileage on the car on each update? 
New updates are coming in quite often, I do not really see the value in showing what the mileage of the car was during each update. And - it is quite easy to manually check this if we want do...

Don't get me wrong; I love all the data Teslamate gives us, and I guess there are still ideas that can be worth adding. But, I think there is some logic behind all the fields/dashboards right now, and it can soon get cluttered with more fields of unknown value :-)

**Comment by SemoTech** [comment]
I beg to differ, it seems to me that as it comes to software updates adding vehicle mileage at the time of update seems a lot more valuable (to me) than "# of charges" or "Average Rated Range".  Also, there is a LOT of space in the "Installed Version" column (as you can see below), so unless it's a big deal I'd love to see mileage added as a column before "Duration" and a shrunk "Installed Version" column.  Thank you.

![image](https://user-images.githubusercontent.com/1452100/151177822-af1339ae-996f-44ab-8010-510c08214c0b.png)

**Comment by sunflower-green** [comment]
I think the number of charges, included "Average rated range" is very useful in the "Updates" dashboard. 
We have seen Tesla change the rated range in a few updates. This specific data in Teslamate shows us exactly that!

But, I wont argue :-)  As you said there is plenty room for more fields if necessary.

**Comment by adriankumpf** [comment]
I'll move this to a discussion as it's feature request not a bug report.
